src/decode.c

```diff
@@ -146,7 +146,7 @@ dwg_decode(Bit_Chain * dat, Dwg_Data * dwg)
   memset(&dwg->header_vars, 0, sizeof(Dwg_Header_Variables));
   memset(&dwg->r2004_header.file_ID_string[0], 0, sizeof(dwg->r2004_header));
   memset(&dwg->auxheader.aux_intro[0], 0, sizeof(dwg->auxheader));
-  memset(&dwg->second_header.size, 0, sizeof(dwg->second_header));
+  memset(&dwg->second_header, 0, sizeof(dwg->second_header));
```

Comment:
I remember vaguely there was a reason to only clear the size. Need to check

Reply:
size is the first field of the structure and sizeof all header. 
I think `&dwg->second_header.size` and `&dwg->second_header are equals`.
Maybe correct code is `memset(&dwg->second_header.size, 0, sizeof(dwg->second_header.size));`?

Reply:
Oh, good call. I didn't see that.

Reply:
Your patch is fine then